### PR TITLE
feat(request-queue): Update RQ listAndLock API with latest API changes

### PR DESCRIPTION
**Pull request by drobnikj**
Update API reference based on latest changes in API.

### CODE REVIEW

**Review by fnesveda — APPROVED** [review]
Looks good, one small improvement

File: apify-api/openapi/components/schemas/request-queues/GetHeadAndLockResponse.yaml [hunk 1]
```diff
@@ -11,7 +11,15 @@ properties:
         example: 1000
       queueModifiedAt:
         type: string
+        description: The modifiedAt the queue is updated whenever the queue is modified. Modifications include adding, updating, or removing requests, as well as locking or unlocking requests in the queue.
```
Comment (janbuchar) [inline apify-api/openapi/components/schemas/request-queues/GetHeadAndLockResponse.yaml:14]: There is probably an extra "the queue" in the first sentence if I'm reading this right. Same in `RequestQueue.yaml`.

Comment (fnesveda) [reply apify-api/openapi/components/schemas/request-queues/GetHeadAndLockResponse.yaml:14]: Ups, I had the same comment but I forgot to send it 😅

File: apify-api/openapi/components/schemas/request-queues/GetHeadAndLockResponse.yaml [hunk 2]
```diff
@@ -11,7 +11,15 @@ properties:
         example: 1000
       queueModifiedAt:
         type: string
+        description: The modifiedAt is updated whenever the queue is modified. Modifications include adding, updating, or removing requests, as well as locking or unlocking requests in the queue.
         example: '2018-03-14T23:00:00.000Z'
+      queueHasLockedRequests:
+        type: boolean
+        description: Whether the queue has locked requests by the same or another client.
```
Comment (janbuchar) [inline apify-api/openapi/components/schemas/request-queues/GetHeadAndLockResponse.yaml:18]:
```suggestion
        description: Whether the queue contains requests locked by any client (either the one calling the endpoint or a different one).
```

**Review by janbuchar — APPROVED** [review]

**Pull request merged**